src/common/utility.cpp

```diff
@@ -84,44 +84,11 @@ QString Utility::formatFingerprint(const QByteArray &fmhash, bool colonSeparated
 
 QString Utility::octetsToString(qint64 octets)
 {
-#define THE_FACTOR 1024
-    static const qint64 kb = THE_FACTOR;
-    static const qint64 mb = THE_FACTOR * kb;
-    static const qint64 gb = THE_FACTOR * mb;
-
-    QString s;
-    qreal value = octets;
-
-    // Whether we care about decimals: only for GB/MB and only
-    // if it's less than 10 units.
-    bool round = true;
-
-    // do not display terra byte with the current units, as when
-    // the MB, GB and KB units were made, there was no TB,
-    // see the JEDEC standard
-    // https://en.wikipedia.org/wiki/JEDEC_memory_standards
-    if (octets >= gb) {
-        s = QCoreApplication::translate("Utility", "%L1 GB");
-        value /= gb;
-        round = false;
-    } else if (octets >= mb) {
-        s = QCoreApplication::translate("Utility", "%L1 MB");
-        value /= mb;
-        round = false;
-    } else if (octets >= kb) {
-        s = QCoreApplication::translate("Utility", "%L1 KB");
-        value /= kb;
-    } else {
-        s = QCoreApplication::translate("Utility", "%L1 B");
-    }
-
-    if (value > 9.95)
-        round = true;
-
-    if (round)
-        return s.arg(qRound(value));
+    OC_ASSERT(octets >= 0)
 
-    return s.arg(value, 0, 'g', 2);
+    using namespace FileSystem::SizeLiterals;
+    const int precesion = quint64(octets) < 1_mb ? 0 : 2;
```

Comment:
No, it was variable, see the removed comment and this test:
```
QCOMPARE(octetsToString(1234567), QString::fromLatin1("1.2 MB"));
```

I'm not sure what to do here. I kind of understand the old behaviour: 117.74 MB (new behaviour) is a bit over-detailed and 118 MB might be better, but the old behaviour isn't very transparent (as your comment shows ;-) ).

Reply:
what we might want to do is: precision 2 for value < 2, precision 1 for value < 10, precision 0 for the rest:
1.23 MB, 3.5 MB, 15 MB

Reply:
@erikjv for context https://chat.owncloud.com/channel/client-dev?msg=9ews6kuTN9L2kSYnR

Reply:
iOS: NSByteCountFormatter with defaults (as used by the iOS app) uses an "adaptive" algorithm:

> The "adaptive" algorithm is platform specific and uses a different number of fraction digits based on the magnitude (in OS X v10.8: 0 fraction digits for bytes and KB; 1 fraction digits for MB; 2 for GB and above). Otherwise the result always tries to show at least three significant digits, introducing fraction digits as necessary.

Reply:
sounds good